packages/page-claims/src/index.tsx

```diff
@@ -73,183 +64,196 @@ const Signature = styled.textarea`
-                />
-                <div>
-                  {t('Copy the above string and sign an Ethereum transaction with the account you used during the pre-sale in the wallet of your choice, using the string as the payload, and then paste the transaction signature object below')}
-                  :
-                </div>
-                <Signature
-                  onChange={this.onChangeSignature}
-                  placeholder={`{\n  "address": "0x ...",\n  "msg": "${prefix}:...",\n  "sig": "0x ...",\n  "version": "2"\n}`}
-                  rows={10}
-                />
-                {(step === Step.Sign) && (
-                  <Button.Group>
-                    <Button
-                      icon='sign-in'
-                      isDisabled={!accountId || !signature}
-                      label={t('Confirm claim')}
-                      onClick={this.setStep(Step.Claim)}
-                    />
-                  </Button.Group>
-                )}
-              </Card>
-            )}
-          </Column>
-          <Column showEmptyText={false}>
-            {(step >= Step.Claim) && (
-              <ClaimDisplay
-                button={this.renderTxButton()}
-                ethereumAddress={ethereumAddress}
-              />
-            )}
-          </Column>
-        </Columar>
-      </main>
-    );
-  }
+  const goToStepSign = useCallback(() => {
+    setStep(Step.Sign);
+  }, []);
 
-  protected isDisabled = (): boolean => {
-    const { accountId, signature } = this.state;
+  const goToStepClaim = useCallback(() => {
+    setStep(Step.Claim);
+  }, []);
 
-    return !accountId || !signature;
-  }
+  // Everytime we get a new preclaimed value (e.g. after we change account), we
+  // decide on which step to show.
+  useEffect(() => {
+    if (isPreclaimed) {
+      goToStepClaim();
+    } else {
+      goToStepSign();
+    }
+  }, [goToStepClaim, goToStepSign, isPreclaimed]);
 
-  protected isUnsigned = (): boolean => true;
+  const onChangeAccount = useCallback((newAccountId) => {
+    setAccountId(newAccountId);
+  }, []);
 
-  protected submitLabel = (): React.ReactNode => this.props.t('Redeem');
+  const onChangeSignature = useCallback((event: React.SyntheticEvent<Element>) => {
+    const { value: signatureJson } = event.target as HTMLInputElement;
 
-  protected txMethod = (): string => 'claims.claim';
+    const { ethereumAddress, signature } = recoverFromJSON(signatureJson);
 
-  protected txParams = (): [string | null, EcdsaSignature | null] => {
-    const { accountId, signature } = this.state;
+    setEthereumAddress(ethereumAddress);
+    setSignature(signature);
+  }, []);
 
-    return [
-      accountId ? accountId.toString() : null,
-      signature || null
-    ];
-  }
+  const onChangeEthereumAddress = useCallback((value: string) => {
+    // FIXME We surely need a better check than just a trim
+    const trimmedAddress = value.trim() as unknown as EthereumAddress;
```

Comment:
`value.trim()` is a string, I think we can either change the state's type to string, or create an actual EthereumAddress here.